Sources/Fullscreen/FavoriteAdsList/FavoriteAdsListView.swift

```diff
@@ -0,0 +1,246 @@
+                             completion: @escaping ((UIImage?) -> Void)
+    )
+    func favoriteAdsListView(_ view: FavoriteAdsListView,
+                             cancelLoadingImageWithPath imagePath: String,
+                             imageWidth: CGFloat
+    )
+}
+
+public class FavoriteAdsListView: UIView {
+
+    // MARK: - Public properties
+
+    public weak var delegate: FavoriteAdsListViewDelegate?
+    public weak var dataSource: FavoriteAdsListViewDataSource?
+
+    public var title: String = "" {
+        didSet { tableHeaderView.title = title }
+    }
+
+    public var subtitle: String = "" {
+        didSet { tableHeaderView.subtitle = subtitle }
+    }
+
+    public var searchBarPlaceholder: String = "" {
+        didSet { tableHeaderView.searchBarPlaceholder = searchBarPlaceholder }
+    }
+
+    public var searchBarText: String {
+        get { return tableHeaderView.searchBarText }
+        set { tableHeaderView.searchBarText = searchBarText }
+    }
+
+    public var sortingTitle: String = "" {
+        didSet { tableHeaderView.sortingTitle = sortingTitle }
+    }
+
+    // MARK: - Private properties
+
+    private let imageCache = ImageMemoryCache()
+    private var didSetTableHeader = false
+
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(withAutoLayout: true)
+        tableView.register(FavoriteAdTableViewCell.self)
+        tableView.register(FavoriteAdsSectionHeaderView.self)
+        tableView.tableFooterView = UIView()
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorInset = .leadingInset(frame.width)
+        tableView.keyboardDismissMode = .onDrag
+        return tableView
+    }()
+
+    private lazy var tableHeaderView: FavoriteAdsListTableHeader = {
+        let tableHeader = FavoriteAdsListTableHeader(withAutoLayout: true)
+        tableHeader.delegate = self
+        tableHeader.searchBarDelegate = self
+        return tableHeader
+    }()
+
+    // MARK: - Init
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) { fatalError() }
+
+    // MARK: - Setup
+
+    private func setup() {
+        addSubview(tableView)
+        tableView.fillInSuperview()
+    }
+
+    // MARK: - Overrides
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if !didSetTableHeader {
+            tableView.tableHeaderView = tableHeaderView
+
+            NSLayoutConstraint.activate([
+                tableHeaderView.topAnchor.constraint(equalTo: tableView.topAnchor),
+                tableHeaderView.centerXAnchor.constraint(equalTo: tableView.centerXAnchor),
+                tableHeaderView.widthAnchor.constraint(equalTo: tableView.widthAnchor)
+            ])
+
+            tableView.tableHeaderView?.layoutIfNeeded()
+            tableView.tableHeaderView = tableView.tableHeaderView
```

Comment:
Why do we set `tableView.tableHeaderView` twice?

Reply:
Again, autolayout for `tableHeaderView` is hard. The view would not be layed out correctly if I didn't

Reply:
Yeah, I remember it was a pain to implement it on the from page 😄